nmoscommon/auth/auth_registrar.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2019 British Broadcasting Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import json
+import requests
+from requests.exceptions import HTTPError
+try:
+    from urlparse import urljoin
+except ImportError:
+    from urllib.parse import urljoin
+
+from ..mdnsbridge import IppmDNSBridge
+
+NMOSOAUTH_DIR = '/var/nmosoauth'
+CREDENTIALS_FILE = 'oauth_credentials.json'
+CREDENTIALS_PATH = os.path.join(NMOSOAUTH_DIR, CREDENTIALS_FILE)
+
+MDNS_SERVICE_TYPE = "nmos-auth"
+
+API_NAMESPACE = 'x-nmos/auth/v1.0/'
+REGISTRATION_ENDPOINT = API_NAMESPACE + 'register_client'
+TOKEN_ENDPOINT = API_NAMESPACE + 'token'
+
+
+def get_credentials_from_file(file):
+    with open(CREDENTIALS_PATH, 'r') as f:
```

Comment:
Probably needs to be some error handling around here to deal with the file being missing